src/ch02-01-values-and-types.md

```diff
@@ -28,17 +28,138 @@ Here, the parameter `a` and the return type must be annotated because it solidif
 usage.
 
+> NOTE: Although the above works, as of version 0.2, Mun is not yet very good at
+>type inferencing. This will be improved in the future.
+
 ### Basic types
 
-Mun knows two basic numeric types: `float` and `int`. A `float` is a
-double-precision IEEE 64-bit floating point number and an `int` represents a
-64-bit integral number.
+Mun understands the following built-in basic value types:
+
+| Type                      | Types
+|---------------------------|---------------------------------------------------|
+| Signed integer types      | `i8`, `i16`, `i32`, `i64`, `i128`, `isize`, `int` |
+| Unsigned integer types    | `u8`, `u16`, `u32`, `u64`, `u128`, `usize`, `uint`|
+| IEEE floating point types | `f32`, `f64`, `float`                             |
+| Boolean type              | `bool` |
+
+Most types are defined by their bitness, so `i8` is an unsigned integer, 8 bits
+in size. There are some exceptions:
+
+* `usize` and `isize` are the size as of the pointer architecture, so 64 bits on
+  64 bit platforms and 32 bits on 32 bit plaforms.
+* `int`, `uint` and `float` are convenience values that are always 64 bits in
+  size. These types are *not* aliases for their sized counterparts so `i64 !=
+  int` and therefor implicit casting is not possible.
+  
+
+### Literals
+
+There are three types of literals in Mun: integer-, floating-point and boolean
+literals. 
+
+A boolean literal is either `true` or `false`.
+
+An integer literal is a number value without a decimal separator (`.`). It can be
+written as an decimal, hexadecimal, octal or binary value. These are all
+examples of valid literals:
+
+```mun
+let a: int = 367;
+let b: int = 0xbeaf;
+let c: int = 0o76532;
+let d: int = 0b0101011;
+```
+
+A floating point literal comes in two forms:
+
+* A decimal number followed by a decimal which is optionally followed by another
+  decimal literal and an *optional* exponent.
+* A decimal number followed by an exponent.
+
+Examples of valid floating point literals are:
+
+```mun
+let a: float = 3.1415;
+let b: float = 3.;
+let c: float = 314.1592654e-2;
+```
+
+#### Separators
+
+Both integer and floating-point literals can contain underscores (`_`) to
+visually separate numbers from one another. They do not have any semantic
+significance but can be useful to the eye.
+
+```mun
+let a: int = 1_000_000;
+let b: float = 1_000.12;
+```
+
+#### Type suffix
+
+Integer and floating-point literals may be followed by a type suffix to
+explicitly specify the type of the literal.
+
+For an integer literal valid suffixes are:
+
+| Literal type | Suffixes |
+|--------------|----------|
+|Integer |`u8`, `i8`, `u16`, `i16`, `u32`, `i32`, `u64`, `i64`, `i128`, `u128`, `int`, `uint`, `usize`, `isize`, `f32`, `f64`, `float` |
+| Floating-point | `f32`, `f64`, `float` |
+
+Note that integer literals can have floating point suffixes. This is not the
+case the other way around.
+
+> Note: number literal types in Mun v0.2 are always either `int` or `float`
```

Comment:
Not relevant to this PR in general, but in regards to a prior discussion that we've had. We decided to keep `int`, `uint` and `float` in the core language spec for now, as we would otherwise not be able to infer the type of a literal without a type suffix. 

Rust actually does this by [always defaulting](https://stackoverflow.com/questions/41996784/what-is-the-integer-or-float-in-a-compiler-error-message) to `i32` for integer types and `f64` for float types. We could do something similar without needing `int`, `uint` and `float`.

Reply:
Yes, I agree. We should remove them at some point.